.github/workflows/ci.yml

```diff
@@ -261,6 +261,12 @@ jobs:
       run: |
         cat /proc/cpuinfo
         ./util/opensslwrap.sh version -c
+    - name: install abigail tools
+      run: sudo apt-get -y install abigail-tools
+    - name: Check libcrypto abi differences
+      run: abidiff ./util/abi/libcrypto-abi.xml ./libcrypto.so
+    - name: Check libssl abi differences
+      run: abidiff ./util/abi/libssl-abi.xml ./libssl.so
```

Comment:
It would make sense to run this in some enable all API features build and have the .xmls in repo covering this. This one has no-ec for example which disables many API calls.

Reply:
Also IMO this does not need to be run in an on-pull request job. On-push would be sufficient IMO.

Reply:
I agree that a separate enable-all build would be a better target for this test, I'm just making sure I can get consistent abilgail diffs right now, will add a spearate build prior to removing draft label here.

As for running on all pull requests, I'm not sure.  I would think we would want to see abi comparisons on all PRs prior to merge to catch abi deltas before they become part of any official branch (though perhaps I'm missing the details of the semantic differences between triggering on PR vs triggering on push).  I guess it depends largely on how compute intensive the test is.

Reply:
I don't see an issue with having this run on PRs. I think it would be desirable.

Reply:
I would not add a separate CI job for it at least. We have quite too many already IMO, but piggybacking on some with possible configure arguments adjustments if needed would be probably ok.

Reply:
one more comment, I am not sure about the placement of the abi.xml files - as they will be relevant only for one particular configure option set I'd place them in the .github or .github/workflows subdirectory so it is clear that this file is not an _official and universal ABI list_

Reply:
I did not mean to add it to multiple jobs. Just amend a single one.

Reply:
That limits the effectivness of the abi check though, as no build has all options enabled, meaning certain abi calls will be excluded from the check

Reply:
I meant add all options that have API implications to one of those builds.
In particular the fips_and_ktls job seems very much suitable for this.
```diff
diff --git a/.github/workflows/ci.yml b/.github/workflows/ci.yml
index 815f7c61b5..87da3f6147 100644
--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -273,7 +273,7 @@ jobs:
     - name: modprobe tls
       run: sudo modprobe tls
     - name: config
-      run: ./config --banner=Configured --strict-warnings enable-ktls enable-fips && perl configdata.pm --dump
+      run: ./config --banner=Configured --strict-warnings enable-ktls enable-fips enable-ec_nistp_64_gcc_128 enable-md2 enable-rc5 enable-sctp enable-ssl3 enable-ssl3-method enable-trace enable-zlib && perl configdata.pm --dump
     - name: make
       run: make -s -j4
     - name: get cpu info
```

Reply:
Perhaps zstd and egd could be also enabled and you'll need to install some packages for zstd and sctp, but you get the idea.

Reply:
yeah, if its acceptable to do that, I'm fine with modifying an existing config

Reply:
ok, I've removed the abi check job and integrated it with the fips/klts test.  For some reason the sslapi test is failing, but I can't see why, I've not modified that test